test/image_test.py

```diff
@@ -523,28 +523,27 @@ def _assertSurfaceEqual(self, surf_a, surf_b, msg=None):
     def test_fromstring__and_tostring(self):
         """Ensure methods tostring() and fromstring() are symmetric."""
 
-        ####################################################################
-        def RotateRGBAtoARGB(str_buf):
-            byte_buf = array.array("B", str_buf)
-            num_quads = len(byte_buf) // 4
-            for i in range(num_quads):
-                alpha = byte_buf[i * 4 + 3]
-                byte_buf[i * 4 + 3] = byte_buf[i * 4 + 2]
-                byte_buf[i * 4 + 2] = byte_buf[i * 4 + 1]
-                byte_buf[i * 4 + 1] = byte_buf[i * 4 + 0]
-                byte_buf[i * 4 + 0] = alpha
-            return tostring(byte_buf)
+        import itertools
 
-        ####################################################################
-        def RotateARGBtoRGBA(str_buf):
+        fmts = ("RGBA", "ARGB", "BGRA")
+        fmt_permutations = list(itertools.permutations(fmts, 2))
+        fmt_combinations = list(itertools.combinations(fmts, 2))
```

Comment:
Nitpicky thing: I believe these can be left as generators, rather than converted to lists.

Reply:
I listified them for debugging purposes (easier to print).
Forgot to remove that :)